archive/walk.go

```diff
@@ -312,3 +312,43 @@ func RmDirs(ctx context.Context, ark *Archive) error {
 		return ark.dataSrc.RemoveAll(ctx, chunk.ZarDir(ark))
 	})
 }
+
+// A SpanInfo is a logical view of the records within a time span, stored
+// in one or more Chunks.
+type SpanInfo struct {
+	Span nano.Span
+
+	// chunks are the data files that contain records within this SpanInfo.
+	// The Chunks may have spans that extend beyond this SpanInfo, so any
+	// records from these Chunks should be limited to those that fall within a
+	// closed span constructed from First & Last.
```

Comment:
```suggestion
	// records from these Chunks should be limited to those that fall within
	// this SpanInfo's Span.
```
And note that spans are always half-open on the right, so "closed span" isn't well-defined..